kubeadm: update all kind(er) jobs for 1.16

neolit123 opened the pull request:
- use prow job annotations
- don't use "latest" kubekins images
- remove 1.12 jobs for kind(er)
- add 1.15 jobs for kind(er)
- don't use "stable" in the names and instead use a version.
existing jobs with "stable" in the name technically should use "ci/latest-x.yy" artifacts
- add kind(er) jobs to release-x.yy-informing dashboards
- update security jobs

/sig cluster-lifecycle
/priority important-longterm
/kind cleanup
/assign @fabriziopandini 
/hold

xref https://github.com/kubernetes/kubeadm/pull/1643

neolit123 commented:
xref https://github.com/kubernetes/test-infra/pull/13205

timothysc commented on config/jobs/kubernetes-security/generated-security-jobs.yaml:
```diff
@@ -506,8 +503,7 @@ presubmits:
         - kubetest
-        image: gcr.io/k8s-testimages/kubekins-e2e:latest-master
-        imagePullPolicy: Always
+        image: gcr.io/k8s-testimages/kubekins-e2e:v20190628-9455316-master
```
Why are we switching this tag and pull policy?

neolit123 replied:
these are debug left overs.

the correct way is to use the appropriate kubekins:
- tagged for a k8s branch
- not always pre-pulling the latest one

timothysc commented on config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml:
```diff
@@ -25,8 +25,7 @@ periodics:
     containers:
-    - image: gcr.io/k8s-testimages/kubekins-e2e:latest-1.12
-      imagePullPolicy: Always
+    - image: gcr.io/k8s-testimages/kubekins-e2e:v20190628-9455316-1.13
```
same

chases2 commented on testgrid/config.yaml:
```diff
@@ -3823,67 +3817,79 @@ dashboards:
+  - name: kubeadm-kind-1.15
+    test_group_name: ci-kubernetes-e2e-kubeadm-kind-1-15
+    alert_options:
+      alert_mail_to_addresses: kubernetes-sig-cluster-lifecycle@googlegroups.com
```
#13205 is a change to migrate everyone to a testgrid annotation, but all of the annotations here are currently supported, [right now](https://github.com/kubernetes/test-infra/blob/master/testgrid/config.md)

If this has a corresponding Prow Job to move it to, then it won't be moved out of this file in the near future

neolit123 replied:
i will have the annotations in this PR.
thanks to the annotations and with some smart (but not so obvious) naming of test_groups and planning we can even avoid touching testgrid/config.yaml for approval, although the current limitations around security job approval for presubmits leaves us with pinging the same folks (top-level approvers).

chases2 commented on testgrid/config.yaml:
```diff
@@ -3823,67 +3817,79 @@ dashboards:
     num_failures_to_alert: 4 # Runs every 2h. Alert when it's been failing for 8 hours
+  - name: kubeadm-kinder-external-etcd-1.15
+    test_group_name: ci-kubernetes-e2e-kubeadm-kinder-external-etcd-1-15
+    alert_options:
```
Similar

neolit123 commented:
@fabriziopandini @chases2 updated with the feedback.
also please take a look at the testgrid -> prow job annotation changes.

chases2 commented:
/lgtm

Thanks!

fabriziopandini commented:
/assign @spiffxp

neolit123 commented:
/assign @cjwagner 
i think @spiffxp might be on vacation(?).

cjwagner commented:
I don't know the full context for these changes, but I suspect we should be using @Katharine's config forker/rotator: https://github.com/kubernetes/test-infra/tree/master/experiment/config-forker#config-forker
https://github.com/kubernetes/test-infra/tree/master/experiment/config-rotator#config-rotator

neolit123 commented:
> I don't know the full context for these changes

OP explains all changes. no automation was applied for these jobs after 1.15 released, so 1.12 should be removed and also jobs had the wrong kubekins.

> but I suspect we should be using @Katharine's config forker/rotator

i wasn't aware of the forker tool, maybe that's the automation we need. but it seems like the PR already does what the tool is designed for (yet the jobs lack the required forker annotations).

not sure what rotator does as the kubeadm jobs are no longer going to use the "stable" labeling.

can the tooling related changes be applied as amends after this PR merges?
we need 1.15 signal ideally ASAP, as this has been up for 7 days.

cjwagner commented:
> i wasn't aware of the forker tool, maybe that's the automation we need. but it seems like the PR already does what the tool is designed for (yet the jobs lack the required forker annotations) + more.
> 
> not sure what rotator does as the kubeadm jobs are no longer going to use the "stable" labeling.

Using the annotations and the "stable" labeling allows for these jobs to created/updated/deleted automatically with each release instead of requiring manual updates like this one that may be forgotten as this one was.

> can the tooling related changes be applied as amends after this PR merges?

Sure, but keep in mind that changing the job name results in uploading job results to a new location so changing the job name cuts off the job history displayed by TestGrid.

/approve

neolit123 commented:
> Using the annotations and the "stable" labeling allows for these jobs to created/updated/deleted automatically with each release instead of requiring manual updates like this one that may be forgotten as this one was.

maybe "stable" has a different meaning here. we moved the jobs away from using e.g. "stable-1.15.txt", which left a gap of untested commits and instead we moved to using "latest-1.15.txt".

> Sure, but keep in mind that changing the job name results in uploading job results to a new location so changing the job name cuts off the job history displayed by TestGrid.

yeah, noted. that's an unfortunate side effect, but the change is required.

neolit123 commented:
/hold cancel
